### PR TITLE
Disconnect user remote connections after logout

### PR DESCRIPTION
**Pull request by tedtoer**
I redid previous PR (https://github.com/rails/actioncable-examples/pull/9) and did PR for actioncable (https://github.com/rails/actioncable/pull/33) for this functional.

Original issue: https://github.com/rails/actioncable-examples/issues/7

### CODE REVIEW

File: app/controllers/concerns/authentication.rb [hunk 1]
```diff
@@ -18,6 +18,8 @@ def authenticate_user(user_id)
     end
 
     def unauthenticate_user
+      ActionCable.server.disconnect(current_user: @current_user)
+
```
Comment (dhh) [inline app/controllers/concerns/authentication.rb]: :scissors: newline.

**Comment by tedtoer** [comment]
@dhh fixed

File: test/controllers/concerns/authentication_concern_test.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    # should disconnect all the connections established for current users
+    user = User.create(name: 'Sticky Fingaz')
+
+    post :create, params: { user_id: user.id }
+
+    ActionCable.server.expects(:disconnect).with(has_entries(current_user: user))
+
+    delete :destroy
+  end
+end
```
Comment (dhh) [inline test/controllers/concerns/authentication_concern_test.rb]: I don't think we really need this test. It just tests that we call a mock we already established ourselves. Let's nix it. Then you can nix the mocha inclusion as well and just have the #disconnect call.

Comment (tedtoer) [inline test/controllers/concerns/authentication_concern_test.rb]: Agree with you, maybe better in future write capybara test, It is exactly the same problem as in issue (login, link to comments, logout, post new comment on opened tab etc.)

Ok, I just nix it now

**Comment by tedtoer** [comment]
@dhh fixed

**Comment by dhh** [comment]
Doesn't seem like the Gemfile.lock changes pertain to this either, right?

**Comment by tedtoer** [comment]
@dhh oops :)
fixed

**Comment by dhh** [comment]
Actually, we do need to bump the Gemfile.lock to tie to the latest actioncable, right?

**Comment by tedtoer** [comment]
@dhh my commits faster your comments bro :D

**Comment by tedtoer** [comment]
@dhh you right, but for it need new PR (Ideally), moment - I'll create it

**Comment by tedtoer** [comment]
@dhh though if we will tie new actioncable in other PR, functional of this PR will not works. May I return updating Gemfile.lock in this PR?

**Comment by dhh** [comment]
We should update the Gemfile.lock whenever we implement a feature that relies on a newer actioncable. So all good.

**Comment by tedtoer** [comment]
@dhh done.
p.s.: thank you for your help me with this children's mistakes, I see you love open source and code :)

**Comment by dhh** [comment]
👌– I had the direction wrong when I told you to yank it. You had it right the first time ;)

**Pull request merged**